### ISSUE TITLE
training issue

### ISSUE THREAD

**Issue by zhao-py** [KuanchihHuang/Reason3D#15]
Great work! 
But I encountered a problem when training the **referring** model. After about 20 epochs of training, it seems the model cannot learn effectively and isn't converging at all.

The validation results after 20 epochs are as follows:
Val result: mIoU/Acc50/Acc25 0.0010/0.0015/0.0026

What could be the reason for this? Approximately how many epochs should it begin to get normal results? (must be 100 epo?) Is it necessary to train on four 3090 GPUs? (It shouldn't be like this, right?)

PS:
Main branch
The experiment was run on two 48GB GPUs with exactly the same configuration as in lavis/projects/reason3d/train/reason3d_scanrefer_scratch.yaml

**Comment by KuanchihHuang** [comment]
Hi! Thanks for your interest in our work. Based on your training configuration, there are some potential issues:

(1) We use 4 GPUs for training. With 2 GPUs, you may need to adjust the batch size or scale the learning rate
  accordingly.

(2) Please make sure the pretrained weights for the point cloud encoder (checkpoints/spf_scannet_512.pth or sp_unet_backbone.pth) are correctly loaded.

(3) Can you also check if the training loss decrease for each epoch?

**Comment by zhao-py** [comment]
> Hi! Thanks for your interest in our work. Based on your training configuration, there are some potential issues:
> 
> (1) We use 4 GPUs for training. With 2 GPUs, you may need to adjust the batch size or scale the learning rate accordingly.
> 
> (2) Please make sure the pretrained weights for the point cloud encoder (checkpoints/spf_scannet_512.pth or sp_unet_backbone.pth) are correctly loaded.
> 
> (3) Can you also check if the training loss decrease for each epoch?

Thank you for your prompt reply! 
**For (1):**
2 GPUs (each 48G) do not work? Interestingly, I found the referring segmentation code can be run on a single 32G GPU, but still fails to learn effectively.

**For (2):**
I've tried these 2 pretrained weights but none of them helps. details as follows:
when loading _sp_unet_backbone.pth_, it shows: 
<img width="1848" height="467" alt="Image" src="https://github.com/user-attachments/assets/3eaf1524-0cd6-4a0f-afd0-f3d82c5cc040" />
when loading _spf_scannet_512.pth_, it shows:
<img width="1843" height="780" alt="Image" src="https://github.com/user-attachments/assets/f489793d-5721-4dca-9fac-fa2840306478" />

**For (3):**
The training loss decreases like:
<img width="397" height="498" alt="Image" src="https://github.com/user-attachments/assets/377c15b4-7df3-4df8-a4ed-ca018e7308c9" />

**Comment by KuanchihHuang** [comment]
Yes, if you are training with fewer GPUs, it may be necessary to modify the batch size or learning rate accordingly. In our setup, the training loss reaches approximately 4.x at epoch 20.

**Comment by zhao-py** [comment]
> Yes, if you are training with fewer GPUs, it may be necessary to modify the batch size or learning rate accordingly. In our setup, the training loss reaches approximately 4.x at epoch 20.

you mean the **batch_size_train** in the YAML is the batch per GPU?

**Comment by KuanchihHuang** [comment]
Yes. On a single GPU.

**Comment by zhao-py** [comment]
> Yes. On a single GPU.

four 3090 GPUs
batch_size_train: 16
**OOM**

**Comment by KuanchihHuang** [comment]
For reference, we are training on 4× NVIDIA A6000 GPUs (48 GB each). If you’re encountering OOM on 4×3090, you may need to either use 8 GPUs, or reduce the total batch size and learning rate and train for more epochs (we haven’t tested these alternative settings ourselves).
